overlord/snapstate/snapstate.go

```diff
@@ -426,21 +418,22 @@ func CurrentInfo(s *state.State, name string) (*snap.Info, error) {
 }
 
 // Get retrieves the SnapState of the given snap.
-func Get(s *state.State, name string, snapst *SnapState) error {
+func Get(s *state.State, name string) (SnapState, error) {
```

Comment:
Returning it by value instead of as a pointer here is a bit atypical, and if we return it as a pointer we'll still need some additional logic as the most natural thing would be to return a nil value when an error happens. This is why I didn't do this change originally.. we'll need to fix the call sites to behave slightly differently.

I suggest just closing this for the time being and putting it aside for when we're not so under the weather in terms of in-progress work on the pipeline.